### PR TITLE
ENH: do not fail with 3d nodes - `preprocess.remove_false_nodes()`

### LINKED ISSUE #657: remove_false_nodes throws an unexpected error
### Describe the problem

When trying to remove false nodes from a [street network obtained from Ordnance Survey](https://www.ordnancesurvey.co.uk/products/os-open-roads) I get the following error:

```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[56], line 1
----> 1 streets_BR = mm.remove_false_nodes(streets_BR)
      2 len(streets_BR)

File [~\AppData\Local\miniconda3\envs\ox\Lib\site-packages\momepy\preprocessing.py:260](http://localhost:8889/lab/tree/OneDrive%20-%20University%20of%20Strathclyde/FB/Birmingham/_scripts/~/AppData/Local/miniconda3/envs/ox/Lib/site-packages/momepy/preprocessing.py#line=259), in remove_false_nodes(gdf)
    256 node_coords = shapely.get_coordinates(target_nodes)
    257 coords = np.array(loop_geom.coords)
    258 new_start = (
    259     node_coords[0]
--> 260     if (node_coords[0] != coords[0]).all()
    261     else node_coords[1]
    262 )
    263 new_start_idx = np.where(coords == new_start)[0][0]
    264 rolled_coords = np.roll(coords[:-1], -new_start_idx, axis=0)

ValueError: operands could not be broadcast together with shapes (2,) (3,)
```

Any solution?

### Steps to reproduce

```
import geopandas as gpd
import momepy as mm
import pandas as pd

sk = gpd.read_file('data/OS Open Roads/SK_RoadLink.shp')
so = gpd.read_file('data/OS Open Roads/SO_RoadLink.shp')
sp = gpd.read_file('data/OS Open Roads/SP_RoadLink.shp')

streets = pd.concat([sk, so, sp], ignore_index=True)

streets = streets.reset_index(drop=True).drop_duplicates('identifier')

bound_BR = gpd.read_file('data/boundary.shp')

streets_BR = streets[streets.within(bound_BR)]

streets_BR = mm.remove_false_nodes(streets_BR)
```

### Versions of your packages

momepy = 0.8.0
geopandas = 1.0.1
libpysal = 4.12.1
networkx = 3.3

### Your operating system

Windows 10

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jGaboardi**
* This PR resolves #657 
* Open to suggests for a better error message.
* [CI failure](https://github.com/pysal/momepy/actions/runs/11465621653/job/31904469874) unrelated to this PR

### CODE REVIEW

File: momepy/preprocessing.py [hunk 1]
```diff
@@ -248,10 +248,16 @@ def remove_false_nodes(gdf):
...
             loop_geom = loops.geometry.iloc[ix]
             target_nodes = nodes.geometry.iloc[node_ix[loop_ix == ix]]
+            if target_nodes.has_z.any():
+                raise ValueError(
+                    "One or more input nodes is 3D. "
+                    "Force 2D with ``geopandas.GeoSeries.force_2d()``."
+                )
             if len(target_nodes) == 2:
                 node_coords = shapely.get_coordinates(target_nodes)
                 coords = np.array(loop_geom.coords)
```
Comment (martinfleis) [inline momepy/preprocessing.py:263]:
```suggestion
            if len(target_nodes) == 2:
                node_coords = shapely.get_coordinates(target_nodes)
                coords = shapely.get_coordinates(loop_geom)
```

Another option is this, which makes the whole thing work. We just drop Z coordinates in that case.

Comment (jGaboardi) [reply momepy/preprocessing.py:263]: So do we think:
* [ ] a check + warning if 3D coords present
* [ ] simply forcing 2D without check

Comment (martinfleis) [reply momepy/preprocessing.py:263]: I would make a note in the docstring and let it use 2d.

File: momepy/preprocessing.py [hunk 2]
```diff
@@ -247,14 +248,16 @@ def remove_false_nodes(gdf):
...
         loops = combined[combined.is_ring]
+
         node_ix, loop_ix = loops.sindex.query(nodes.geometry, predicate="intersects")
         for ix in np.unique(loop_ix):
             loop_geom = loops.geometry.iloc[ix]
             target_nodes = nodes.geometry.iloc[node_ix[loop_ix == ix]]
             if len(target_nodes) == 2:
                 node_coords = shapely.get_coordinates(target_nodes)
-                coords = np.array(loop_geom.coords)
+                coords = np.array(force_2d(loop_geom).coords)
```
Comment (martinfleis) [inline momepy/preprocessing.py:260]:
```suggestion
                coords = shapely.get_coordinates(loop_geom)
```
No need to create a new (2D) geometry and then fetch its coordinates.

**Comment by martinfleis** [comment]
pre-commit.ci autofix

**Pull request merged**